src/test/java/de/tum/in/www1/artemis/service/FileServiceTest.java

```diff
@@ -311,4 +311,17 @@ void testDeleteFiles_shouldNotThrowException() {
         assertDoesNotThrow(() -> fileService.deleteFiles(List.of(path)));
         mockedFiles.close();
     }
+
+    @Test
+    void testRenameDirectory() {
+        // File is deleted automatically after test
+        writeFile("some-path/some-file", "some-content");
+        Path source_dir = Path.of("./exportTest/some-path");
+        Path target_dir = Path.of("./exportTest/some-path2");
+        Path target_file = Path.of("./exportTest/some-path2/some-file");
```

Comment:
```suggestion
        Path sourceDir = Path.of("./exportTest/some-path");
        Path targetDir = Path.of("./exportTest/some-path2");
        Path targetFile = Path.of("./exportTest/some-path2/some-file");
```

Reply:
Yes I agree, there are also still usages of strings in the public API.
We should duplicate the method header, and mark it as `@deprecated`, then use delegation to delegate to the Method with the signature we want to be used. i.e.

```java
/**
 * @deprecated. use method ...
 */
public void methodToBeRefactored(String path) {
  methodToBeRefactored(Path.from(path));
}

public void methodToBeRefactored(Path path) {
  // Logic moved here.
}

```

Reply:
In case it wasn't clear. This was just a suggestion how I would phase out deprecated code

Reply:
So I should focus merely on the API of the FileService and leave the responsibility for calling the correct method to the caller ( e.g another service ) ?

Reply:
> While this improves the APi of the FileService, by just calling `toFile` we still use the old API inside of the method. This should also be changed to use the `nio` methods.